ts/src/pro/delta.ts

```diff
@@ -0,0 +1,451 @@
+         * @param {string} symbol unified symbol of the market to fetch the order book for
+         * @param {int} [limit] the maximum amount of order book entries to return
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/#/?id=order-book-structure} indexed by market symbols
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        symbol = market['symbol'];
+        const messageHash = 'book:' + symbol;
+        const subscriptionHash = 'l2_updates';
+        const request = {
+            'type': 'subscribe',
+            'payload': {
+                'channels': [
+                    {
+                        'name': subscriptionHash,
+                        'symbols': [
+                            market['id'],
+                        ],
+                    },
+                ],
+            },
+        };
+        const url = this.urls['api']['ws'];
+        const orderbook = await this.watch (url, messageHash, this.extend (request, params), messageHash);
+        return orderbook.limit ();
+    }
+
+    handleOrderBook (client: Client, message) {
+        //
+        //  snapshot
+        //     {
+        //         "action": "snapshot",
+        //         "asks": [
+        //             [ "2914.60", "0.1733" ],
+        //             ...
+        //         ],
+        //         "bids": [
+        //             [ "2913.40", "0.6131" ],
+        //             ...
+        //         ],
+        //         "cs": 1905080047,
+        //         "sequence_no": 937625,
+        //         "symbol": "ETH_USDT",
+        //         "timestamp": "1708514696898474",
+        //         "type": "l2_updates"
+        //     }
+        //
+        //  update
+        //     {
+        //         "action": "update",
+        //         "asks": [
+        //         [ "2914.60", "0" ],
+        //         ...
+        //         ],
+        //         "bids": [
+        //         [ "2913.60", "0.6131" ],
+        //         ...
+        //         ],
+        //         "cs": 1145868018,
+        //         "sequence_no": 937626,
+        //         "symbol": "ETH_USDT",
+        //         "timestamp": "1708514697899524",
+        //         "type": "l2_updates"
+        //     }
+        //
+        const action = this.safeString (message, 'action');
+        const marketId = this.safeString (message, 'symbol');
+        const symbol = this.safeSymbol (marketId);
+        const timestamp = this.safeInteger (message, 'timestamp');
+        const messageHash = 'book:' + symbol;
+        let orderbook = this.safeDict (this.orderbooks, symbol) as any;
+        if (orderbook === undefined) {
+            orderbook = this.orderBook ({});
+        }
```

Comment:
Instead of this, now we should do

```
                if (!(symbol in this.orderbooks)) {
                    const ob = this.orderBook ({});
                    this.orderbooks[symbol] = ob;
                }
                const orderbook = this.orderbooks[symbol];
```

By doing that, we preserve the type and it's easier to see the methods available